src/Lib.hs

```diff
@@ -335,21 +335,30 @@ inspectAttachments ticketInfo attachments = runMaybeT $ do
     lastAttachment  <- MaybeT . pure $ lastAttach
     att             <- MaybeT $ getAttachment lastAttachment
 
-    pure $ inspectAttachment config ticketInfo att
+    catchAny (inspectAttachment config ticketInfo att)
+        (\_ ->
+            -- Exception handling, return response stating the log was corruptedqq
+            return $ ZendeskResponse
+                { zrTicketId    = (tiId ticketInfo)
+                , zrComment     = prettyFormatLogReadError ticketInfo
+                , zrTags        = TicketTags [renderTicketStatus CannotReadZipFile]
+                , zrIsPublic    = (cfgIsCommentPublic config)
+                }
+        )
 
 
 -- | Given number of file of inspect, knowledgebase and attachment,
 -- analyze the logs and return the results.
-inspectAttachment :: Config -> TicketInfo -> AttachmentContent -> ZendeskResponse
+inspectAttachment :: (MonadCatch m) => Config -> TicketInfo -> AttachmentContent -> m ZendeskResponse
```

Comment:
Don't touch this, this is all pure, no need to infect that part of the code with impurities.

Reply:
Got it!